src/server/scriptInfo.ts

```diff
@@ -398,14 +398,14 @@ namespace ts.server {
                 if (!this.formatSettings) {
                     this.formatSettings = getDefaultFormatCodeSettings(this.host);

```

Comment:
This function returns a copy, so it is ok to modify .. So no need to clone unlike defaultPreferences... I am not sure how many times we call these configure function to say if its ok to clone the object (esp with such a big set of properties)

Reply:
Done.